e2e/capacity/provider_boost.test.ts

```diff
@@ -0,0 +1,46 @@
+import '@frequency-chain/api-augment';
+import assert from 'assert';
+import { getFundingSource } from '../scaffolding/funding';
+import {
+  createKeys,
+  createMsaAndProvider,
+  CENTS,
+  DOLLARS,
+  createAndFundKeypair,
+  boostProvider,
+} from '../scaffolding/helpers';
+
+const fundingSource = getFundingSource('capacity-provider-boost');
+
+describe('Capacity: provider_boost extrinsic', function () {
+  const providerBalance = 2n * DOLLARS;
+
+  it('happy path succeeds', async function () {
+    const stakeKeys = createKeys('booster');
+    const provider = await createMsaAndProvider(fundingSource, stakeKeys, 'Provider1', providerBalance);
+    const booster = await createAndFundKeypair(fundingSource, 5n * DOLLARS, 'booster');
+    await assert.doesNotReject(boostProvider(fundingSource, booster, provider, 1n * DOLLARS));
+  });
+
+  it('fails when they are a Maximized Capacity staker', async function () {
+    const stakeKeys = createKeys('booster');
+    const provider = await createMsaAndProvider(fundingSource, stakeKeys, 'Provider1', providerBalance);
+    await assert.rejects(boostProvider(fundingSource, stakeKeys, provider, 1n * DOLLARS));
+  });
+
+  it("fails when they don't have enough token", async function () {
```

Comment:
To make sure the test is failing for the right reason (not enough tokens), you can update it to check the specific error message or code. This way, you know it’s not just rejecting for some random reason.

I recommend doing this to all these test so we can ensure we have not introduced a bug.

Reply:
For on-chain extrinsic errors, I believe you can check the code. For RPC (ie, pre-dispatch) errors, we get a generic error code, but I like matching the error message against a RegEx (just some keywords or something, so it is not as brittle if the exact wording of the error changes)